spec/features/application/web_server_spec.rb

```diff
@@ -33,4 +17,37 @@
 
     expect(page.status_code).to eq 403
   end
+
+  scenario 'content security policy' do
+    visit_200_page '/home'
+
+    expect(response_headers['Content-Security-Policy']).to eq [
+      "default-src 'none'",
+      "connect-src 'self'",
+      "font-src 'self' https:",
+      "img-src 'self' https: data:",
+      "script-src 'self' https: 'unsafe-inline'",
+      "style-src 'self' https: 'unsafe-inline'"
+    ].join('; ') + ';'
+  end
+
+  context 'with gzip' do
+    before do
+      page.driver.header('ACCEPT_ENCODING', 'gzip, deflate')
+    end
+
+    scenario 'visiting a page' do
+      visit_200_page '/home'
+
+      expect(response_headers['Content-Encoding']).to eq 'gzip'
+    end
+
+    scenario 'visiting an asset with gzip' do
+      asset_file_path = Dir.glob(Rails.root.join('public/assets/application*.css')).first
+      asset_file_path.gsub!(Rails.root.join('public').to_s, '')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.